### PR TITLE
Adding a completion for $jarsigner  *.apk files

### PR DESCRIPTION
**Pull request by hugoziviani**
Adding a completion for `jarsigner` with `*.apk` files. This bug emerged from the downstream Debian bash-completion. The bug identifier is: **Debian Bug report logs - #907635**

### CODE REVIEW

File: completions/cryptsetup [hunk 1]
```diff
@@ -66,7 +66,7 @@ _cryptsetup()
                         ;;
                 esac
                 ;;
-            luksFormat|luksAddKey|luksRemoveKey)
+            luksFormat|luksAddKey|luksRemoveKey|luksChangeKey)
```
Comment (scop) [inline completions/cryptsetup]: Unrelated change, please remove.

File: completions/jarsigner [hunk 2]
```diff
@@ -25,33 +25,34 @@ _jarsigner()
...
             return
             ;;
     esac
 
     # Check if a jar was already given.
     local i jar=false
     for (( i=0; i < ${#words[@]}-1; i++ )) ; do
-        if [[ "${words[i]}" == *.jar && \
-            "${words[i-1]}" != -signedjar ]]; then
+        if [[ "${words[i]}" == *.jar || "${words[i]}" == *.apk && \
```
Comment (scop) [inline completions/jarsigner]: Could combine the two conditions to `*.@(jar|apk)`

File: completions/jarsigner [hunk 3]
```diff
@@ -25,33 +25,34 @@ _jarsigner()
...
+            "${words[i-1]}" != -signedjar ]] ; then
             jar=true
             break
         fi
     done
 
-    if ! $jar ; then
-        if [[ "$cur" == -* ]]; then
+    if ! ($jar || $apk) ; then
```
Comment (scop) [inline completions/jarsigner]: Doesn't seem anytihing sets `$apk` nor is necessary, so this change should just be removed.

File: completions/jarsigner [hunk 4]
```diff
@@ -25,33 +25,34 @@ _jarsigner()
...
+        if [[ "$cur" == -* ]] ; then
```
Comment (scop) [inline completions/jarsigner]: Unrelated/undesired whitespace change, please remove.

File: completions/jarsigner [hunk 5]
```diff
@@ -25,33 +25,34 @@ _jarsigner()
...
                 -- "$cur") )
         fi
-        _filedir jar
+        _filedir '@(jar|apk)'
     fi
 } &&
 complete -F _jarsigner jarsigner
 
 # ex: filetype=sh
+
```
Comment (scop) [inline completions/jarsigner]: Unrelated/undesired whitespace change, please remove.

**Comment by hugoziviani** [comment]
Hello @scop , Sorry to be late with the corrections. Thanks to help me correct the patch. Here go the corrections as you asked.

File: completions/jarsigner [hunk 6]
```diff
@@ -25,16 +25,16 @@ _jarsigner()
...
             ;;
     esac
 
     # Check if a jar was already given.
     local i jar=false
     for (( i=0; i < ${#words[@]}-1; i++ )) ; do
-        if [[ "${words[i]}" == *.jar && \
-            "${words[i-1]}" != -signedjar ]]; then
+        if [[ "${words[i]}" == *.@(jar|apk) && \
+            "${words[i-1]}" != -signedjar ]] ; then
```
Comment (scop) [inline completions/jarsigner]: Unneeded/undesired whitespace change still here.

File: completions/jarsigner [hunk 7]
```diff
@@ -49,9 +49,9 @@ _jarsigner()
...
                 -- "$cur") )
         fi
-        _filedir jar
+        _filedir '@(jar|apk)'
     fi
 } &&
 complete -F _jarsigner jarsigner
 
-# ex: filetype=sh
+# ex: filetype=sh
```
Comment (scop) [inline completions/jarsigner]: Unneeded/undesired whitespace change introduced here.

File: completions/jarsigner (hunk 6 above)
Comment (hugoziviani) [reply completions/jarsigner]: Thanks! cheers friend.
